### PR TITLE
Add support for CaDiCaL

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Comment by tautschnig** [comment]
Marking do-not-merge until I have provided data (and cleaned up the patch series as it does additional work that should go in a separate PR).

**Comment by tautschnig** [comment]
On a randomly selected set of 53 Dimacs problems generated from SV-COMP benchmarks, CaDiCal solves 46 within 900 seconds while Minisat with simplifier solves 28 and without simplifiers solves 38.

File: src/solvers/sat/satcheck_cadical.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  virtual tvt l_get(literalt a) const;
+
+  virtual void lcnf(const bvt &bv);
+  virtual void set_assignment(literalt a, bool value);
+
+  virtual void set_assumptions(const bvt &_assumptions);
+  virtual bool has_set_assumptions() const { return false; }
+  virtual bool has_is_in_conflict() const { return false; }
+  virtual bool is_in_conflict(literalt a) const;
+
```
Comment (kroening) [inline src/solvers/sat/satcheck_cadical.h]: we would probably now have an "override" for all of the above.

**Comment by kroening** [comment]
Add the override, rebase, ready to go!

**Comment by tautschnig** [comment]
@kroening Overrides added, rebased.

**Pull request merged**